Makefile

```diff
@@ -286,18 +288,18 @@ ifeq ($(DEV_IMAGE), true)
 # the dist directory is under .dockerignore.
 IMAGE_TAG="dev-$(shell git describe --always --dirty)"
 image: build-ui
-	DOCKER_BUILDKIT=1 docker build -t argocd-base --target argocd-base .
+	DOCKER_BUILDKIT=1 docker build ${DOCKER_PLATFORM} -t argocd-base --target argocd-base .
 	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -v -ldflags '${LDFLAGS}' -o ${DIST_DIR}/argocd ./cmd
```

Comment:
So I am not sure how much value it makes in being able to configure DOCKER_PLATFORM because currently Dockerfile.dev works by building bins locally and then just copying them over and we currntly hard code golang's build configuration to linux/amd64.

As far as docker for desktop is concerned it can run multi arch images by default https://docs.docker.com/build/building/multi-platform/#support-on-docker-desktop so I think that we could also just hardcode docker --platform as well or make both configurable.

Reply:
I see your point. TBH, I would just hardcode both in this case.